### ISSUE TITLE
explorer do nor work with filter

### ISSUE THREAD

**Issue by pat1** [ARPA-SIMC/dballe#218]
```
#!/usr/bin/python3
import dballe

# Create from database
db = dballe.DB.connect("sqlite:db.sqlite")
with dballe.Explorer("explorer") as explorer:
    with explorer.rebuild() as updater:
        with db.transaction() as tr:
            updater.add_db(tr)
            #updater.set_filter({"ident":"agrmet"})
    print ("created from DB")
    print (explorer.all_reports)
    print (explorer.all_levels)
    print (explorer.all_tranges)
    print (explorer.all_varcodes)
    print (explorer.stats)

with dballe.Explorer("explorer") as explorer:
    filter={"report":"simnpr"}    
    explorer.set_filter(filter)
    print ("explorer filtered by : ",filter)
    print (explorer.reports)
    print (explorer.levels)
    print (explorer.tranges)
    print (explorer.varcodes)
```

```
python explorerfromdb.py
created from DB
['agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'agrmet', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'fixed', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'locali', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'mnw', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'profe', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnbo', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'simnpr', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'spdsra', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane', 'urbane']
[dballe.Level(103,2000,None,None)]
[dballe.Trange(254,0,0)]
['B12101']
ExplorerStats(datetime_min=datetime.datetime(2020, 7, 3, 0, 0), datetime_max=datetime.datetime(2020, 7, 3, 14, 0), count=8212)
explorer filtered by :  {'report': 'simnpr'}
[]
[]
[]
[]

```
db used for test:
[explorer.zip](https://github.com/ARPA-SIMC/dballe/files/4874637/explorer.zip)

**Comment by spanezz** [comment]
Quick update: I reproduced it in the test suite. It looks like a xapian-specific problem, I'll dig further.

**Comment by spanezz** [comment]
Fixed in c1c7e2249a540a793320f594419d25da314387cb (it turns out one cannot close multiple issues in the same commit message with that syntax)
